sklearn/tests/test_cross_validation.py

```diff
@@ -696,6 +696,16 @@ def test_train_test_split():
     assert_equal(split[2].shape, (7, 7, 11))
     assert_equal(split[3].shape, (3, 7, 11))
 
+    # test stratification option
+    y = np.array([1, 1, 1, 1, 2, 2, 2, 2])
+    for test_size, exp_test_size in zip([2, 4, 0.25, 0.5, 0.75],
+                                        [2, 4, 2, 4, 6]):
+        train, test = cval.train_test_split(y,
```

Comment:
you should fix the random state of ``train_test_split``. Also, I'm not entirely convinced of the test. I guess it would fail with ``stratify=None``, though.